pkgs/applications/misc/gpu-burn/default.nix

```diff
@@ -1,45 +1,64 @@
-{ lib, stdenv, fetchFromGitHub, addOpenGLRunpath, cudatoolkit }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, symlinkJoin
+, cudaPackages
+}:
+let
+
+  cudapath = symlinkJoin {
+    name = "cuda-deps";
+    paths = with cudaPackages; [
+      cuda_cudart
+      cuda_nvcc
+      libcublas
+    ];
+  };
```

Comment:
Looking at the readme, we could drop `symlinkJoin` and pass `makeFlags = [ "LDFLAGS=-L..." "CFLAGS=-I... NVCCFLAGS=-ccbin..." ]` instead. This way we probably would eliminate e.g. nvcc from the runtime closure

I don't mind the current way though

Reply:
I'll try it. It effectively rewrites half of the (admittedly small) Makefile, but it is better.

Reply:
I've tried this but `gpu_burn` is failing with:

```
Couldn't init a GPU test: Error in load module (gpu_burn-drv.cpp:234): PTX JIT compiler library not found
```

I'm doing this:

```nix
  makeFlags = [
    "CUDAPATH=/does-not-exist"
    "NVCC=${cuda_nvcc}/bin/nvcc"
    "NVCCFLAGS=-I${cuda_cudart}/include"
  ];

  CFLAGS = [
    "-I${cuda_cudart}/include"
    "-I${cuda_nvcc}/include"
    "-I${libcublas}/include"
  ];

  LDFLAGS = [
    "-L${cuda_cudart}/lib"
    "-L${cuda_cudart}/lib/stubs"
    "-L${libcublas}/lib"
  ];
```

If you have any tips, let me know. When I `strace` the working version, there is no mention of a "ptx jit" compiler so I'm not sure what's missing.

Reply:
> `strace`

`LD_DEBUG=libs gpu_burn` will be probably less noisy

Reply:
I found the problem. TIL that `make VAR=3` will completely override `VAR` in the `Makefile`, even if that `VAR` has `VAR += ...` entries in the Makefile. BUT, `VAR=3 make` will *not* completely override it and will cause the `VAR += ...` in the Makefile to append to the `3` you specified. This mistake caused `NVCCFLAGS` to be wrong in my example above. I still have not found the exact place in the Make documentation that describes this alarming "feature."

Reply:
ux 😍